<!DOCTYPE html>
<html>
<head>
  	<meta charset="utf-8" />
  	<meta http-equiv="X-UA-Compatible" content="IE=edge" />
  	<meta name="viewport" content="width=device-width, initial-scale=1" />
  
<link rel="stylesheet" href="https://mattecapu.github.io/ct-zulip-archive/style.css" /><title>biology and reaction networks · theory: science · Zulip Chat Archive</title>
</head>
<body>
<header>
<a href="https://mattecapu.github.io/ct-zulip-archive" class="home-link">
        <img class="logo" src="https://zulip-avatars.s3.amazonaws.com/21317/realm/icon.png?version=3" />
        <h1>Category Theory<br/>Zulip Server<br/>Archive</h1>
        </a>
        <p>
        You're reading the public-facing archive of the <a href="https://categorytheory.zulipchat.com/">Category Theory Zulip server</a>.<br/>
        
        To join the server you need an invite. Anybody can get an invite by contacting <a href="https://matteocapucci.wordpress.com">Matteo Capucci</a> at <em>name dot surname at gmail dot com</em>.<br/>
        
        For all things related to this archive refer to the same person.
        </p>
        </header>
        <hr />
    
<h2>Stream: <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/index.html">theory: science</a></h2>
<h3>Topic: <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html">biology and reaction networks</a></h3>

<hr>

<base href="https://categorytheory.zulipchat.com">

<a name="543601722"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/543601722" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Joe Moeller <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#543601722">(Oct 07 2025 at 18:54)</a>:</h4>
<p><span class="user-mention silent" data-user-id="275920">John Baez</span> <a href="#narrow/channel/411259-theory.3A-science/topic/DNA/near/543167608">said</a>:</p>
<blockquote>
<p>I don't know a definition of "program" other than the definition in computer science!    Maybe there is one.... okay, yes, you can ask for the program at a concert.  </p>
</blockquote>
<p>There's also non/linear programming as in optimization, the program(me) of a conference, a person's research program. My wife was involved in film festivals for a while and I was surprised when she talked about programming. My grandpa called TV shows "programs". Obviously the unifying thing is they're all about detailed instructions/planning. The only objection I could really get is that "program" seems to have a connotation of intention, somehow even more than "code" does, which doesn't exactly make sense for DNA. Well, unless you're talking about synthetic biology, which I think we might be!</p>



<a name="543697556"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/543697556" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> John Baez <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#543697556">(Oct 08 2025 at 09:16)</a>:</h4>
<p>Okay, I call TV shows programs - maybe only old people do that?   But I don't think bacteria have TV shows going on inside them.   <span aria-label="upside down" class="emoji emoji-1f643" role="img" title="upside down">:upside_down:</span> </p>
<p>In computer science I think there's a difference between an algorithm, and  a program (which is a specific realization of an algorithm in some language), and what the physical material of the computer is doing when it's running the program.  There are probably even more layers of abstraction (assembly language, etc.).  I'm no expert on the ontology of computer science; someone must have thought a lot about this.</p>
<p>Biology probably involves <em>different</em> layers of abstraction: there's no reason to assume it works the same way.   So if we're trying to understand the world of life, we shouldn't just transplant the concepts from computer science unthinkingly.    </p>
<p>In synthetic biology we are, of course, free to impose any concepts we want, since we're designing new systems.   There we are only limited by <em>what works</em>.</p>
<p>Personally I'm more interested in understanding the living world that came before us than synthetic biology.    I think we have a very crude attitude toward technology, and we have a lot to learn from the living world if we don't flatten it out.   It's been around for a lot longer than us, and it does amazing things we can barely fathom.   Right now we're busy pushing the world toward an <a href="https://en.wikipedia.org/wiki/Holocene_extinction">mass extinction event</a>, which would be the first extinction event on Earth caused by a supposedly intelligent organism that messes with the biosphere too much before understanding it.</p>



<a name="543699335"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/543699335" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> John Baez <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#543699335">(Oct 08 2025 at 09:25)</a>:</h4>
<p>Anyway, I'm interested in this paper by Corentin:</p>
<ul>
<li><a href="https://arxiv.org/abs/2001.02116v5">Ergodicity, Output-Controllability, and Antithetic Integral Control of Uncertain Stochastic Reaction Networks</a>.</li>
</ul>
<p>and would be happy to talk about it!</p>



<a name="543708146"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/543708146" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#543708146">(Oct 08 2025 at 10:11)</a>:</h4>
<p>I am happy to answer any question regarding this paper. It is an extension of the results in the Cell Systems paper I previously mentioned.</p>



<a name="544142309"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/544142309" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#544142309">(Oct 10 2025 at 11:06)</a>:</h4>
<p><span class="user-mention silent" data-user-id="275920">John Baez</span> <a href="#narrow/channel/411259-theory.3A-science/topic/biology.20and.20reaction.20networks/near/543160382">said</a>:</p>
<blockquote>
<p>I like how your second paper blends control theory with the study of reaction networks.  I wrote a <a href="http://math.ucr.edu/home/baez/stoch_stable.pdf">book</a> on reaction networks and Petri nets... not using category theory, but <a href="http://math.ucr.edu/home/baez/RxNet.pdf">later</a> I started working with them using category theory.   I feel there's a lot left to be done, but people seem to be having trouble knowing how to start.  Maybe if we talk here you can come up with something good.</p>
</blockquote>
<p>I am reading at the moment the book on reaction networks and Petri Nets. I may have a few questions at some point but I am waiting to be towards the end to ask them. I am almost done with the book.</p>



<a name="544153208"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/544153208" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> John Baez <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#544153208">(Oct 10 2025 at 12:06)</a>:</h4>
<p>Great!  I don't have any intelligent questions about your paper yet,  but I'm going to want to know how you prove ergodicity.</p>



<a name="544177613"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/544177613" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#544177613">(Oct 10 2025 at 14:03)</a>:</h4>
<p>We construct Foster-Lyapunov functions and show irreducibility of the state-space. It does not cover all open reaction networks, only unimolecular mass-action networks, certain classes of unimolecular mass-action networks, and certain classes of bimolecular networks.</p>



<a name="544177770"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/544177770" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#544177770">(Oct 10 2025 at 14:04)</a>:</h4>
<p>This relies on results by Meyn and Tweedie.</p>



<a name="544178454"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/544178454" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#544178454">(Oct 10 2025 at 14:07)</a>:</h4>
<p>Our earlier paper on the topic: <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003669">https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003669</a> and the paper of Meyn and Tweedie: <a href="https://meyn.ece.ufl.edu/wp-content/uploads/sites/77/archive/spm_files/Papers_pdf/meytweIII.pdf">https://meyn.ece.ufl.edu/wp-content/uploads/sites/77/archive/spm_files/Papers_pdf/meytweIII.pdf</a></p>



<a name="544179577"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/544179577" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#544179577">(Oct 10 2025 at 14:12)</a>:</h4>
<p>If this is not clear, I am happy to provide more details or answer any questions.</p>



<a name="544179868"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/544179868" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#544179868">(Oct 10 2025 at 14:13)</a>:</h4>
<p>Regarding the PLoS paper, all the interesting stuff is in the Supplementary Information <a href="https://doi.org/10.1371/journal.pcbi.1003669.s001">https://doi.org/10.1371/journal.pcbi.1003669.s001</a></p>



<a name="544382710"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/544382710" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#544382710">(Oct 12 2025 at 08:41)</a>:</h4>
<p><span class="user-mention" data-user-id="275920">@John Baez</span> I am now done with the book. It was a very nice read. I was quite familiar with a lot of the contents, so it went fast but I learned quite a few new things that will certainly be useful in the near future, I am sure. I also got a bunch of new references I was not aware of. I will need to go through them soon.</p>



<a name="544385665"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/544385665" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> John Baez <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#544385665">(Oct 12 2025 at 09:32)</a>:</h4>
<p>Great!</p>



<a name="553342798"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553342798" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553342798">(Nov 03 2025 at 10:09)</a>:</h4>
<p><span class="user-mention" data-user-id="275920">@John Baez</span> <span class="user-mention" data-user-id="520375">@Adittya Chaudhuri</span> The workshop I mentioned the other time is approaching (Nov. 19). We will post the full program on the website  <a href="https://www.biocontrolseminars.org/workshop-2025">https://www.biocontrolseminars.org/workshop-2025</a>. Maybe some of the talks could be of interest to you. Some will be technical, some more experimental.</p>



<a name="553377906"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553377906" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Adittya Chaudhuri <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553377906">(Nov 03 2025 at 13:14)</a>:</h4>
<p>Thanks very much <span class="user-mention" data-user-id="970092">@Corentin Briat</span>. I just registered. Looking forward to attending the talks on November 19th.</p>



<a name="553379918"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553379918" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Adittya Chaudhuri <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553379918">(Nov 03 2025 at 13:24)</a>:</h4>
<p>I was reading about <a href="https://ieeexplore.ieee.org/document/5462880">Boolean control networks</a> for some days. I find the concept interesting (although I need some more time to learn more  concretely). From the point of "<em>controlling gene regulatory networks</em>", it would be great to get some overview on the present state of research on the applications of  Boolean control networks in gene regulatory networks.</p>



<a name="553385810"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553385810" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553385810">(Nov 03 2025 at 13:51)</a>:</h4>
<p>Great, thanks for your interest. If you face any issues or anything else, feel free to contact me directly.</p>



<a name="553390695"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553390695" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553390695">(Nov 03 2025 at 14:12)</a>:</h4>
<p>Boolean networks are elegant mathematical/theoretical constructs, but they can sometimes be too abstract or detached from practical biological applications. From a purely theoretical standpoint, significant work has been published on the analysis and control of Boolean networks, for instance <a href="https://link.springer.com/book/10.1007/978-0-85729-097-7">Analysis and Control of Boolean Networks</a></p>



<a name="553413506"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553413506" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Adittya Chaudhuri <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553413506">(Nov 03 2025 at 15:38)</a>:</h4>
<p>Thank you.</p>



<a name="553476986"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553476986" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Ryan Wisnesky <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553476986">(Nov 03 2025 at 20:44)</a>:</h4>
<p>I'm surprised the idea of computation as re-writing didn't appear on this thread.  A lot of CS theory models a program as a collection of rewrite rules, for example, 1+1 rewrites to 2 and x+0 rewrites to x, etc.  (Even CQL turns category presentations into directed re-write rules.)   That looks a lot to me like chemistry: H2 + O rewrites to H2O, etc.  So it seems to me like you can get a very basic correspondence between chemistry and computation simply by considering "weights" in re-writing theory.</p>



<a name="553487100"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553487100" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Jacques Carette <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553487100">(Nov 03 2025 at 21:49)</a>:</h4>
<p>The downside of basing computation on rewriting is that for many cases, it ends up in the wrong complexity class. Polynomial arithmetic being an easy example.</p>
<p>That's not to say that rewriting isn't a good model for physical theories where physical proximity is important (including biology and chemistry in 'physical').</p>



<a name="553499158"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553499158" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Peva Blanchard <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553499158">(Nov 03 2025 at 23:30)</a>:</h4>
<p><span class="user-mention silent" data-user-id="281326">Ryan Wisnesky</span> <a href="#narrow/channel/411259-theory.3A-science/topic/biology.20and.20reaction.20networks/near/553476986">said</a>:</p>
<blockquote>
<p>I'm surprised the idea of computation as re-writing didn't appear on this thread.  A lot of CS theory models a program as a collection of rewrite rules, for example, 1+1 rewrites to 2 and x+0 rewrites to x, etc.  (Even CQL turns category presentations into directed re-write rules.)   That looks a lot to me like chemistry: H2 + O rewrites to H2O, etc.  So it seems to me like you can get a very basic correspondence between chemistry and computation simply by considering "weights" in re-writing theory.</p>
</blockquote>
<p>A long time ago I stumbled upon the <a href="https://kappalanguage.org/">Kappa language</a> which seems to fit this description. They also mention <a href="https://github.com/RuleWorld/bionetgen">Bionetgen</a> and <a href="https://cheminf.imada.sdu.dk/mod/">Mød</a>. I don't know if these projects are still active today.</p>



<a name="553499462"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553499462" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> John Baez <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553499462">(Nov 03 2025 at 23:33)</a>:</h4>
<p><span class="user-mention" data-user-id="405160">@ww</span> does simulations in Kappa, and he's trying to adapt some of those ideas to the AlgebraicJulia software (based on category theory) that our team is using, like <a href="https://github.com/AlgebraicJulia/AlgebraicRewriting.jl">AlgebraicRewriting.jl</a>.   This is the sort of thing we talk about a lot.</p>



<a name="553566911"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553566911" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Nathaniel Virgo <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553566911">(Nov 04 2025 at 09:46)</a>:</h4>
<p>MØD is still active for sure, there's a steady stream of papers using it.</p>



<a name="553729562"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553729562" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> ww <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553729562">(Nov 04 2025 at 22:39)</a>:</h4>
<p>Kappa is alive and well. I started out using it for synthetic biology but during the acute phase of the pandemic found that it is well suited to epidemics. We also found ourselves chafing against some design choices (such as the "rigidity" property of port graphs that makes the expensive subgraph isomorphism checking much, much cheaper) which is a good part of my motivation for working on AlgebraicRewriting.jl. Eventually I would like to have a Kappa-like language front end for it which should get a really nice balance of expressivity and efficiency.</p>



<a name="553855581"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553855581" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> ww <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553855581">(Nov 05 2025 at 13:32)</a>:</h4>
<p><span class="user-mention silent" data-user-id="275920">John Baez</span> <a href="#narrow/channel/411259-theory.3A-science/topic/biology.20and.20reaction.20networks/near/543163833">said</a>:</p>
<blockquote>
<p>You say "the program", but I explained that there's no "program" in a bacterial cell: a "program" as we normally understand it is a symbol string that directs something to carry out a precise sequence of instructions.</p>
</blockquote>
<p>I am also used to thinking of a DNA (or RNA) sequence like a program, it certainly is a symbol string and the action of proteins like RNA polymerase on the sequence is to slide along and read it and execute the instructions by producing proteins. But it is a fundamentally parallel process, it is noisy and it can have crosstalk. So quite different from our well-behaved silicon computers. And the nucleic acid computational gadget interacts with other computational gadgets in a cell like the protein-protein interactions.</p>



<a name="553914487"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553914487" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> John Baez <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553914487">(Nov 05 2025 at 17:02)</a>:</h4>
<p>I think the mechanism of the cell is so fundamentally different that considering it a "noisy" version of our "well-behaved" computers is pretty misleading.... it can fool us into thinking the strangely different features of biology are suboptimal, while neglecting that cells need to survive under vastly more changeable conditions than a computer sitting on a shelf - and they need, most of all, to reproduce!</p>



<a name="553923336"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553923336" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553923336">(Nov 05 2025 at 17:38)</a>:</h4>
<p>Interesting read on the topic: Wetware: A Computer in Every Living Cell by Dennis Bray.</p>



<a name="553982364"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553982364" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> ww <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553982364">(Nov 05 2025 at 23:50)</a>:</h4>
<p>I certainly agree that we should not be fooled like that. The biggest difference to me is that there is no high-level language and the program, having been evolved rather than written, is unlike any program a human might write.</p>
<p>I am reminded of an experiment that someone (I forget who, I'll try to find the paper) did in the 1990s with an FPGA board and a genetic algorithm. An FPGA is a kind of computer chip where you can rewire the gates in software. They set an objective which was to react to some input (maybe it was temperature) and produce some output (maybe it was turn on or off the heating or turn on a light or something) and set the genetic algorithm running. The resulting hardware configuration was nothing like an engineer would do. No modularity and separation into logical blocks or anything like that. The circuit that developed used odd effects like stray capacitance between adjacent paths which engineers try to avoid and we basically pretend do not happen.</p>
<p>Maybe I am mangling the story a bit because it's been a long time since I read the paper, but that's the general idea.</p>
<p>So I think similarly about DNA and cellular processes: they truly are computational devices but the way they are programmed (or rather have programmed themselves in response to evolutionary pressures) is radically different from our computer science.</p>



<a name="553983043"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/553983043" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> ww <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#553983043">(Nov 05 2025 at 23:57)</a>:</h4>
<p>"An evolved circuit, intrinsic in silicon, entwined with physics" by Adrian Thompson <a href="https://link.springer.com/chapter/10.1007/3-540-63173-9_61">https://link.springer.com/chapter/10.1007/3-540-63173-9_61</a></p>
<p>(Non-paywall version: <a href="https://gwern.net/doc/ai/1997-thompson.pdf">https://gwern.net/doc/ai/1997-thompson.pdf</a>)</p>
<p>The task was tone discrimination.</p>



<a name="554013071"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/554013071" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#554013071">(Nov 06 2025 at 06:27)</a>:</h4>
<p><span class="user-mention" data-user-id="405160">@ww</span> Many papers have been published recently that report curious circuit designs using genetic algorithms or similar, and researchers are trying now to decypher them and understand their structure. As you mention, those circuits do not seem to exhibit any clear modular structure,  as is the case of a cell for instance, where everything is pretty much coupled with everything via cross-talk and resource sharing. The lack of modularity is certainly one of the main challenges in understanding how a cell works. From an engineering perspective, we also lack the tools to create non-modular designs, that would achieve a certain function under some constraints, such as cost/ energy constraints, in a rather monolithic manner. This is something I have been thinking about for quite some time now.</p>



<a name="554139957"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/554139957" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Adittya Chaudhuri <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#554139957">(Nov 06 2025 at 16:44)</a>:</h4>
<p><span class="user-mention silent" data-user-id="970092">Corentin Briat</span> <a href="#narrow/channel/411259-theory.3A-science/topic/biology.20and.20reaction.20networks/near/554013071">said</a>:</p>
<blockquote>
<p>From an engineering perspective, we also lack the tools to create non-modular designs, that would achieve a certain function under some constraints, such as cost/ energy constraints, in a rather monolithic manner. This is something I have been thinking about for quite some time now.</p>
</blockquote>
<p>Can we think of non-modular design as a kind of <em>integrated framework</em>?</p>
<p>For example, these papers:</p>
<p><a href="https://ieeexplore.ieee.org/document/6160954">Integration of large-scale metabolic, signaling, and gene regulatory networks with application to infection responses</a> by Richard et al.</p>
<p><a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1000086">Dynamic Analysis of Integrated Signaling, Metabolic, and Regulatory Networks</a> by Lee et al.</p>
<p>talked about <em>integrating</em> signalling pathways, metabolic networks and gene regulatory networks into a single framework to achive something (eg: infection response).</p>



<a name="554178233"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/554178233" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Kevin Carlson <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#554178233">(Nov 06 2025 at 19:54)</a>:</h4>
<p><span class="user-mention silent" data-user-id="405160">ww</span> <a href="#narrow/channel/411259-theory.3A-science/topic/biology.20and.20reaction.20networks/near/553982364">said</a>:</p>
<blockquote>
<p>I certainly agree that we should not be fooled like that. The biggest difference to me is that there is no high-level language and the program, having been evolved rather than written, is unlike any program a human might write.</p>
</blockquote>
<p>Surely it's also a central point that biological function is not actually a mathematical function of the DNA sequence? I'm not at all solid on the biology, but my impression is that this has been one of the big struggles in maximizing the impact of modern genomic science: even if you know what protein a gene codes for, you still have to figure out its geometry, and then there's all kinds of RNA and epigenetics and things that also complicate the computational metaphor.</p>



<a name="554247776"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/554247776" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#554247776">(Nov 07 2025 at 06:42)</a>:</h4>
<p><span class="user-mention" data-user-id="520375">@Adittya Chaudhuri</span> I would make a distinction. Integrated does not mean non-modular. For instance, in the second paper you mention, the authors clearly distinguish multiple modules (regulatory, metabolic, and signaling). In this regard, the process they study is viewed as an interconnection of modules. Non-modular, would mean that would have the same system with the same functionality, however we would not be able to clearly distinguish or separate the process into submodules, each with their on clear functionality that could be disconnected and studied on its own. This is essentially equivalent to saying that a reductionist approach does not apply and that this biological system should be approach using a holistic approach.</p>



<a name="554249452"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/554249452" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#554249452">(Nov 07 2025 at 06:58)</a>:</h4>
<p><span class="user-mention" data-user-id="609515">@Kevin Carlson</span>  I am not sure to understand your point. The fact that we have difficulties predicting three dimensional structures of proteins based on their amino acid sequence is certainly one problem but this is an independent one. In any way, whether there is a mathematical function between codons and proteins and functionality or not, it does not change the fact that there is a mapping, whatever that mapping is, and biology knows and exploits that. Functionality does not stop at proteins but is also implemented at the network level and RNA/epigenetics does not complicate at all the "computational metphor", rather supports it. Epigenetics is about regulation and so are some types of mRNA such as small RNAs, microRNAs. As a side note, a non-negligible part of the DNA is human is non-coding, meaning that it does not code for proteins, and its functionality is regulatory in essence. For instance, proteins called transcription factors can bind to DNA at certain sites to modulate the activity of a gene. In the end a cell is all about computation and decision making. There are production chains (metabolic networks) to produce molecules required for the cell the survive or house keeping stuffs to make sure that everything is recycled. Oscillators are present to measure time and make decisions based on that. There are receptors (sensors) all over the place on the membrane that measure the environment and this information is transmitted to the inner working of the cells through signaling networks for the cell to make an appropriate decision, such as for instance, switching carbon source, apoptosis, compensate for shock, etc.</p>



<a name="554295351"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/554295351" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Adittya Chaudhuri <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#554295351">(Nov 07 2025 at 11:07)</a>:</h4>
<p><span class="user-mention silent" data-user-id="970092">Corentin Briat</span> <a href="#narrow/channel/411259-theory.3A-science/topic/biology.20and.20reaction.20networks/near/554247776">said</a>:</p>
<blockquote>
<p>This is essentially equivalent to saying that a reductionist approach does not apply and that this biological system should be approach using a holistic approach.</p>
</blockquote>
<p>Thanks!! I got your point now.  </p>
<p>For example,  one may study the time-varying gene expression data (find correlations between various gene expressions) with respect to a set of signals to a cell.  One may use various methods of data analysis for the same. One may obtain certain conclusions "about how various genes are coexpressed with respect to various signals".  Now, one may try to build an integrated model comprising signalling pathways, metabolic networks and gene regulatory networks associated with the given signals entering the cell  in such a way that dynamics of the integrated model matches the "conclusion of the data analysis" (upto certain extent) when we fit the model with the required biolgical data. Of course, this model is not perfect as it depends on a certain hypothesis <em>to match the output of the model</em> with the  biological conclusion coming from the conclusion of the said data analysis of gene expressions. Can we think of this model as a model built from an holistic approach? If yes, then,  other than starting from data, what are some approaches to model cellular decision making which can be considered as a holistic approach?</p>



<a name="554461518"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/554461518" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#554461518">(Nov 08 2025 at 07:34)</a>:</h4>
<p><span class="user-mention" data-user-id="520375">@Adittya Chaudhuri</span>  This is a very good question. I do not have a definitive answer, only an opinion.</p>
<p>In general, a holistic approach means analyzing a system as a whole rather than as a collection of independent modules that can be studied separately, which would correspond to reductionism.</p>
<p>Let us consider a simple unicellular system such as a bacterium. A truly holistic approach would involve studying the entire cell directly, without decomposing it into subsystems that are analyzed in isolation. However, if the organism exhibits some form of modular structure, then analyzing those modules separately may be acceptable, provided that their environment is also incorporated into the model in some way to account for what we call context dependence.</p>
<p>It is not unreasonable to assume that living systems display some degree of modularity, and this is precisely the assumption made when we study specific mechanisms or pathways, myself included. This simplification is necessary because a fully holistic approach remains far beyond our reach. We currently lack the biological understanding, computational resources, and mathematical tools required to model a living system in its entirety.</p>
<p>Regarding your question, I would say that under certain weak modularity assumptions, where the overall signaling, metabolic, and regulatory networks are only loosely coupled to the rest of the cell, such an analysis could be considered holistic in a broad sense. However, this is rarely the case in practice and probably never will be completely true. Still, we need to start somewhere.</p>
<p>Ultimately, data are the only direct source of information we have about biological processes. We can try to build phenomenological or mechanistic models to represent these data and attempt to reverse engineer what is happening. However, we often face fundamental limitations, such as the lack of predictive power: these models frequently fail to predict the outcome of new experiments or the emergence of previously unobserved behaviors, even qualitatively.</p>
<p>If the goal of a model is simply to reproduce existing data, then yes, we know how to do that, and in fact, the vast majority of papers published in reputable biology journals report exactly this kind of modeling.</p>



<a name="554483370"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/554483370" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Adittya Chaudhuri <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#554483370">(Nov 08 2025 at 13:37)</a>:</h4>
<p><span class="user-mention" data-user-id="970092">@Corentin Briat</span>  Thanks  very much for your explanation on the holistic approach.</p>



<a name="554484181"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/554484181" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Adittya Chaudhuri <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#554484181">(Nov 08 2025 at 13:50)</a>:</h4>
<p><span class="user-mention silent" data-user-id="970092">Corentin Briat</span> <a href="#narrow/channel/411259-theory.3A-science/topic/biology.20and.20reaction.20networks/near/554461518">said</a>:</p>
<blockquote>
<p>Ultimately, data are the only direct source of information we have about biological processes. We can try to build phenomenological or mechanistic models to represent these data and attempt to reverse engineer what is happening. However, we often face fundamental limitations, such as the lack of predictive power: these models frequently fail to predict the outcome of new experiments or the emergence of previously unobserved behaviors, even qualitatively.</p>
<p>If the goal of a model is simply to reproduce existing data, then yes, we know how to do that, and in fact, the vast majority of papers published in reputable biology journals report exactly this kind of modeling.</p>
</blockquote>
<p>I agree.  </p>
<p>If we consider a Boolean model of a  gene regulatory network (although it forgets many details) and study the properties of the attractors to conclude something on the behaviour of the cell and <em>only then</em> we test our hypothesis on some data set, can we count the approach as a holistic approach  in a trivial way (as we have ignored the existence of any other  modules like signalling networks,  metabolic networks,..etc) ? Thus, in a way (if I am not misunderstanding), the model of the cell reduces to the Boolean model of the gene regulatory network in the cell .</p>



<a name="554587364"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/554587364" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Corentin Briat <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#554587364">(Nov 09 2025 at 19:48)</a>:</h4>
<p><span class="user-mention" data-user-id="520375">@Adittya Chaudhuri</span> I am not sure to understand your question. But it is difficult a priori to say if a approach is "holistic enough" in the case of biology. But if you assume that the content of the cell is just regulatory networks and model that as a whole, then yes I would say that it would count as a holistic approach. In reality, that will not be the case, though.</p>



<a name="554749612"></a>
<h4><a href="https://categorytheory.zulipchat.com#narrow/stream/411259-theory%3A%20science/topic/biology%20and%20reaction%20networks/near/554749612" class="zl"><img src="https://mattecapu.github.io/ct-zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Adittya Chaudhuri <a href="https://mattecapu.github.io/ct-zulip-archive/stream/411259-theory.3A-science/topic/biology.20and.20reaction.20networks.html#554749612">(Nov 10 2025 at 16:58)</a>:</h4>
<p>Thanks. I am getting your points.</p>



<footer class="site-footer">

<hr><p>Last updated: Feb 28 2026 at 12:12 UTC</p>
This archive runs on a customization of <a href="https://github.com/zulip/zulip-archive">zulip-archive</a>
</footer>
</body>

</html>